lib/web/fetch/index.js

```diff
@@ -135,114 +191,115 @@ function fetch (input, init = undefined) {
   try {
     requestObject = new Request(input, init)
   } catch (e) {
-    p.reject(e)
-    return p.promise
+    return Promise.reject(e)
```

Comment:
1 microtask delayed.

Reply:
@tsctx I am not sure what you mean by this, can you elaborate?

Reply:
@crysmags I think it could be implemented with fewer changes. In the way... https://github.com/tsctx/undici/commit/d7740e1732dc3d6eef93b4080e818b4b2d961c84

Reply:
`new Promise((_, reject) => reject(v))` is rejected synchronously, while `Promise.reject(v)` is rejected asynchronously (after one microtask). It must be rejected synchronously as per the spec.

Reply:
That was my misunderstanding. I mistake `Promise.resolve` for `Promise.reject`. In which case there is no difference.
FYI `Promise.resolve(promise)` allows the same `promise` to be returned. but, `new Promise(resolve => resolve(promise))` creates a new Promise and resolves to 1 microtask.